device/iot-device-samples/android-sample/app/src/main/java/com/iothub/azure/microsoft/com/androidsample/MainActivity.java

```diff
@@ -331,48 +161,34 @@ public IotHubMessageResult execute(Message msg, Object context)
         }
     }
 
-    static class EventCallback implements IotHubEventCallback
+    static class MessageSentCallbackImpl implements MessageSentCallback
     {
-        public void execute(IotHubStatusCode status, Object context)
+        @Override
+        public void onMessageSent(Message message, IotHubClientException e, Object o)
         {
-            Integer i = (Integer) context;
-            System.out.println("IoT Hub responded to message " + i.toString()
-                    + " with status " + status.name());
+            if (e == null)
+            {
+                System.out.println("IoT Hub responded to message " + message.getMessageId() + " with status OK");
+            }
+            else
+            {
+                System.out.println("IoT Hub responded to message " + message.getMessageId() + " with status " + e.getStatusCode().name());
+            }
         }
     }
 
     static class MessageCallback implements com.microsoft.azure.sdk.iot.device.MessageCallback
     {
-        public IotHubMessageResult execute(Message msg, Object context)
+        public IotHubMessageResult onCloudToDeviceMessageReceived(Message msg, Object context)
         {
             Counter counter = (Counter) context;
             System.out.println(
                     "Received message " + counter.toString()
                             + " with content: " + new String(msg.getBytes(), Message.DEFAULT_IOTHUB_MESSAGE_CHARSET));
 
-            int switchVal = counter.get() % 3;
```

Comment:
Not sure why we ever wanted to randomly decide what to do with received messages